Services/KeyLoader/AbstractKeyLoader.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Services\KeyLoader;
+
+/**
+ * Load configuration keys.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+abstract class AbstractKeyLoader implements KeyLoaderInterface
+{
+    /**
+     * @var string
+     */
+    protected $privateKey;
+
+    /**
+     * @var string
+     */
+    protected $publicKey;
+
+    /**
+     * @var string
+     */
+    protected $passphrase;
+
+    /**
+     * Constructor.
+     *
+     * @param string $privateKey
+     * @param string $publicKey
+     * @param string $passphrase
+     */
+    public function __construct($privateKey, $publicKey, $passphrase)
+    {
+        $this->privateKey = $privateKey;
+        $this->publicKey  = $publicKey;
+        $this->passphrase = $passphrase;
+    }
+
+    /**
+     * Checks that configured keys exists and private key can be parsed using the passphrase
+     */
+    public function checkConfig()
+    {
+        $this->loadKey('public');
+        $this->loadKey('private');
+    }
+
+    /**
+     * @param string $type The key type, must be "public" or "private"
+     *
+     * @return string The path of the key
+     *
+     * @throws \InvalidArgumentException If the given type is not valid.
+     */
+    public function getKeyPath($type)
```

Comment:
It looks like this should really be protected, except that a few tests rely on it. We should at least mark it with an `@internal` - it's not meant to be used from the outside.